cmd/neofs-node/config.go

```diff
@@ -705,16 +708,54 @@ func (c *cfg) shardOpts() []shardOptsWithID {
 		for _, sRead := range shCfg.subStorages {
 			switch sRead.typ {
 			case blobovniczatree.Type:
+				ppdPath := filepath.Join(filepath.Dir(sRead.path), "peapod.db")
+				ppd := peapod.New(ppdPath, sRead.perm)
+
+				_, err := os.Stat(ppdPath)
+				if err == nil {
+					c.log.Info("data was already migrated from Blobovnicza tree to Peapod, continue with it")
+					ss = append(ss, blobstor.SubStorage{
+						Storage: ppd,
+						Policy: func(_ *objectSDK.Object, data []byte) bool {
+							return uint64(len(data)) < shCfg.smallSizeObjectLimit
+						},
+					})
+					break
+				}
+
+				if !errors.Is(err, fs.ErrNotExist) {
+					fatalOnErrDetails("get info about Peapod file", err)
+				}
+
+				var compressCfg compression.Config
+				compressCfg.Enabled = shCfg.compress
+				compressCfg.UncompressableContentTypes = shCfg.uncompressableContentType
+
+				err = compressCfg.Init()
+				fatalOnErrDetails("init compression config", err)
+
+				bbcz := blobovniczatree.NewBlobovniczaTree(
+					blobovniczatree.WithRootPath(sRead.path),
+					blobovniczatree.WithPermissions(sRead.perm),
+					blobovniczatree.WithBlobovniczaSize(sRead.size),
+					blobovniczatree.WithBlobovniczaShallowDepth(sRead.depth),
+					blobovniczatree.WithBlobovniczaShallowWidth(sRead.width),
+					blobovniczatree.WithOpenedCacheSize(sRead.openedCacheSize),
+					blobovniczatree.WithLogger(c.log))
+				bbcz.SetCompressor(&compressCfg)
+
+				ppd.SetCompressor(&compressCfg)
+
+				c.log.Info("configured Blobovnicza tree is deprecated, overtaking data into Peapod...",
```

Comment:
So we're going to have the same configuration with `blobovnicza`, but using peapod? It seems a bit too magic to me. How about keeping bbcz as is, but creating a tool
```
blobovnicza_to_peapod [config]
```
that would:
 * do all of this logic inside
 * rewrite config to have proper data for peapod (drop bbcz, add ppd)

This would allow to:
 * work with bbcz as is for a release, nothing will break for sure
 * migrate config along with data
 * check that everything is OK on one node and then decide on subsequent migration
 * continue using bbcz if migration fails for some reason

Then we could move bbcz code into the tool itself and keep it for some time (with no support in the node).

Reply:
nah, to complex to me. Imo we should migrate bolder. As an altenative I suggest:
1. fallback to bbcz tree automatically here if `Copy` failed
2. leave blobovnicza config flag `no_peapod` to force the use of bbcz (for example, we migrated successfully, but the data is not available from peapod)

so, if migration fails, we see the reason and node works with bbcz. If migration succeeds, node works with peapod. And if we later see that some data disappeared after migration, we set config flag manually and work with bccz until peapod is fixed.  

thoughts?

Reply:
It's too much magic for an admin to handle, I think. It should be more predictable:
 * I have a config and I run a new node version -> I expect it to work according to the config
 * I'm doing migration with a tool, it succeeds -> I have a new config and data, all good
 * I'm doing migration with a tool, it fails -> I can use an old config and data, all good
 * I'm doing migration with a tool, it succeeds, but the node fails to start afterwards -> I take the old config and use it

Compare that to:
 * I run a new node version and pray for it to work
 * I have no idea what's going on with my data unless I'm constantly looking at the log
 * I have some config which specifically says 'blobovnicza', but I need to look at the log to see what's really used internally

In general, automigration is cool, but it'd be more possible if we were to have a configuration like in #2497 (with all internal fstree/bbcz relations hidden). While we have some `/dir/bbcz` config and will get `/dir/peapod.db` file (that technically can be in some wrong (not intended to be used) directory.

Reply:
where to place this migration tool? new `cmd/`?

Reply:
Yeah, it'd be a simple new tool.